crypto/ocsp/ocsp_client.c

```diff
@@ -247,3 +248,61 @@ int OCSP_resp_find_status(OCSP_BASICRESP *bs, OCSP_CERTID *id, int *status,
   }
   return 1;
 }
+
+int OCSP_check_validity(ASN1_GENERALIZEDTIME *thisUpdate,
+                        ASN1_GENERALIZEDTIME *nextUpdate,
+                        long drift_num_seconds, long max_age_seconds) {
+  int ret = 1;
+  int64_t t_tmp;
+  int64_t t_now = time(NULL);
+
+  // Check |thisUpdate| is valid and not more than |drift_num_seconds| in the
+  // future.
+  if (!ASN1_GENERALIZEDTIME_check(thisUpdate)) {
```

Comment:
Is my understanding correct that the ASN.1 GeneralizedTime is expected to be UTC only, and not have any other timezone offset?

Reply:
Great question, I looked into a bit more and the RFC states that "GeneralizedTime values MUST be expressed in Greenwich Mean Time", but can optionally "include a representation of the time differential between local and Greenwich Mean Time".
* https://datatracker.ietf.org/doc/html/rfc6960#section-4.2.2.1
* https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.5.2

So the time is expected to UTC only (no local time allowed), but a differential offset can be included.

Reply:
This definition is just in the context of CRLs and OCSP